lib/model-builder.js

```diff
@@ -308,7 +308,9 @@ ModelBuilder.prototype.define = function defineClass(className, properties, sett
   // Add the id property
   if (idInjection) {
     // Set up the id property
-    ModelClass.definition.defineProperty('id', {type: Number, id: 1, generated: true});
+    // updateOnly property is added to indicate that this property will appear in the model for update/updateorcreate
+    // operations but and not for create operation.
+    ModelClass.definition.defineProperty('id', {type: Number, id: 1, generated: true, updateOnly: true});
```

Comment:
@bajtos We can't set updateOnly property in the above place, since  LoopBack repository's PersistentModel.setupRemoting() --> getUpdateOnlyProperties() gets invoked before above logic in datasource-gluggler's datasource.js code gets executed. Since getUpdateOnlyProperties() gets invoked before setting 'updateOnly' model as you suggested above, getUpdateOnlyProperties() will not return id property even if forceId is true. Any other parts of the logic we can set this property?